src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -596,19 +599,16 @@ impl Net {
                 .add(tx_queue.len(mem).into());
             let head_index = head.index;
             // Parse IoVecBuffer from descriptor head
-            let buffer = match IoVecBuffer::from_descriptor_chain(head) {
-                Ok(buffer) => buffer,
-                Err(_) => {
-                    self.metrics.tx_fails.inc();
-                    tx_queue
-                        .add_used(mem, head_index, 0)
-                        .map_err(DeviceError::QueueError)?;
-                    continue;
-                }
+            if unsafe { self.io_vec_buffer.load_descriptor_chain(head).is_err() } {
+                self.metrics.tx_fails.inc();
+                tx_queue
+                    .add_used(mem, head_index, 0)
+                    .map_err(DeviceError::QueueError)?;
+                continue;
             };
 
             // We only handle frames that are up to MAX_BUFFER_SIZE
-            if buffer.len() as usize > MAX_BUFFER_SIZE {
+            if self.io_vec_buffer.len() as usize > MAX_BUFFER_SIZE {
```

Comment:
Actually, can we avoid it when clippy isn't forcing us to? u32/u64 -> usize is always safe, and I really dislike the clutter that `try_into().unwrap()` unnecessarily adds :sweat_smile:

Reply:
I know it's very unlikely to happen, but I like the idea of eliminating the possibility of silent lossy conversions. Again, I know a lossy conversion is not possible here, but I'd like us to avoid the habit.

Ideally, I'd like us to have something like that: https://github.com/materializeinc/materialize/blob/main/src/ore/src/cast.rs.

But anyways, this is a nit, no need to apply.